app/src/main/kotlin/com/ustadmobile/zim2xapi/AddxAPIStatementUseCase.kt

```diff
@@ -0,0 +1,45 @@
+package com.ustadmobile.zim2xapi
+
+import org.jsoup.Jsoup
+import java.io.File
+
+class AddxAPIStatementUseCase {
+
+    operator fun invoke(
+        zimFolder: File,
+        passingGrade: Int
+    ) {
+
+        val perseusFile = File(zimFolder, "/assets/perseus/perseus_script.js")
+
+        // if doesn't exist, presume not a khan exercise
+        if (!perseusFile.exists()) return
+
+        val indexFile = File(zimFolder, "index.html")
+
+        val indexDoc = Jsoup.parse(indexFile)
+        val scoreScript = indexDoc.createElement("script")
+        scoreScript.attr("src", "./assets/perseus/score-tracker.js")
+            .attr("type", "module")
+        val widgetsScript = indexDoc.createElement("script")
+        widgetsScript.attr("src", "./assets/perseus/khan-widgets.js")
+            .attr("type", "module")
+        indexDoc.body().appendChild(scoreScript).appendChild(widgetsScript)
+
+        indexFile.writeText(indexDoc.html())
+
+        val passingGradeDeclaration = "const passingGrade = $passingGrade;\n"
+        val scoreFile = File(zimFolder, "/assets/perseus/score-tracker.js")
+        val scoreText = this::class.java.classLoader.getResource("score-tracker.js")?.readText() ?: ""
```

Comment:
I recommend against using ?.readText here. If for whatever reason score-tracker.js did not load - then that is a fatal error. It should throw an exception, not just silently ignore it.

Reply:
done throws Exceptions